src/index.ts

```diff
@@ -60,85 +64,244 @@ export interface DynamoDBAutoIncrementProps {
-            ':nextCounter': nextCounter,
-          },
-          Key: this.props.counterTableKey,
-          TableName: this.props.counterTableName,
-          UpdateExpression: 'SET #counter = :nextCounter',
+        if (counter === undefined) {
+          nextCounter = this.props.initialValue + 1
+          Update = {
+            ConditionExpression: 'attribute_not_exists(#counter)',
+            ExpressionAttributeNames: {
+              ...ExpressionAttributeNames,
+              '#counter': this.#attributeName,
+            },
+            ExpressionAttributeValues: {
+              ...ExpressionAttributeValues,
+              ':nextCounter': nextCounter,
+            },
+            Key: this.props.counterTableKey,
+            TableName: this.props.tableName,
+            UpdateExpression,
+          }
+        } else {
+          nextCounter = counter + 1
+          Update = {
+            ConditionExpression: '#counter = :counter',
+            ExpressionAttributeNames: {
+              ...ExpressionAttributeNames,
+              '#counter': this.#attributeName,
+            },
+            ExpressionAttributeValues: {
+              ...ExpressionAttributeValues,
+              ':counter': counter,
+              ':nextCounter': nextCounter,
+            },
+            Key: this.props.counterTableKey,
+            TableName: this.props.tableName,
+            UpdateExpression,
+          }
         }
-      } else {
-        nextCounter = counter + 1
-        Update = {
-          ConditionExpression: '#counter = :counter',
+
+        // PUT should be done to this.props.counterTableName
+        const Put: PutCommandInput = {
+          ConditionExpression: `${this.#getKeyConditionExpression(
+            this.props.counterTableKey
+          )} AND attribute_not_exists(#counter)`,
           ExpressionAttributeNames: {
-            '#counter': this.props.counterTableAttributeName,
+            ...this.#getKeyExpressionAttributeNames(this.props.counterTableKey),
+            '#counter': this.props.tableAttributeName,
           },
-          ExpressionAttributeValues: {
-            ':counter': counter,
-            ':nextCounter': nextCounter,
+          Item: {
+            ...previousItem,
+            [this.props.tableAttributeName]: counter ?? 1,
           },
-          Key: this.props.counterTableKey,
           TableName: this.props.counterTableName,
-          UpdateExpression: 'SET #counter = :nextCounter',
         }
-      }
 
-      const Put: PutCommandInput = {
-        ConditionExpression: 'attribute_not_exists(#counter)',
-        ExpressionAttributeNames: { '#counter': this.props.tableAttributeName },
-        Item: { [this.props.tableAttributeName]: nextCounter, ...item },
-        TableName: this.props.tableName,
+        if (this.props.dangerously) {
+          const temp = (
+            await this.props.doc.query({
+              KeyConditionExpression: 'widgetID = :widgetID',
```

Comment:
What's with the attribute name `widgetID` hardcoded here?

Reply:
this was from a log test, removed it in the latest commit